src/lib/image.tsx

```diff
@@ -0,0 +1,56 @@
+import { config } from '../config';
+import cdn, { ImageFetchOptions } from './cdn';
+
+interface SourceOptions {
+  src: string;
+  local?: boolean;
+  options: ImageFetchOptions;
+}
+
+export function getSourceUrl(src: string) {
+  if (src && src.includes('__zer0ImageCrop__=')) {
+    try {
+      let nonParams, queryParams, _ignore;
+      [_ignore, nonParams, queryParams] = src.match(/(.*)\?(.*)/);
+
+      queryParams = new URLSearchParams(queryParams);
+      queryParams.delete('__zer0ImageCrop__');
+
+      if (queryParams && [...queryParams.keys()].length !== 0) {
+        return [nonParams, '?', queryParams.toString()].join('');
+      }
+
+      return nonParams;
+    } catch (error) {
+      console.log('error parsing source', error, src);
+    }
+  }
+
+  return src;
+}
+
+export function getSource({ src, local = false, options = {} }: SourceOptions) {
+  let source = src;
+
+  if (!local) {
+    if (source.indexOf('/assets/') >= 0 || source.indexOf('data:image') === 0) {
+      return source;
+    }
+
+    source = getSourceUrl(source);
+
+    const cloudinaryRegex = new RegExp(`https:\/\/res.cloudinary.com\/${config.cloudinary.cloud_name}\/(image|video)\/upload\/v[0-9]+\/`, 'i');
```

Comment:
related to the config thing above, i don't think we should be assuming cloudinary at the level of a global module. instead, let's encapsulate this in a provider that's specific to the background image component. in the image component itself we can work to an interface. that lets us swap out cloudinary easily, and contextually in the future if we want, and also solves the problem of providing configuration.

eg.
```
// components/background-image/index.ts in zOS:

import { CloudinaryProvider, BackgroundImage as bImage } from 'zos-component-library';

export function BackgroundImage() {
  const provider = new CloudinaryProvider({ name: 'my-cloud-name' });
  return (
    <bImage
      {...this.props}
      provider={provider}
    />;
}
```